drivers/at86rf2xx/aes_spi.c

```diff
@@ -0,0 +1,432 @@
+        _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+        _aes_transfer_bytes(dev, true, first, NULL, AT86RF2XX_AES_BLOCK_SIZE);
+        _aes_transfer_bytes(dev, false, &mirror, NULL, sizeof(mirror));
+
+        xtimer_usleep(AT86RF2XX_AES_DELAY_US);
+        if ((status = _aes_check(dev)) & AT86RF2XX_AES_STATUS_MASK__AES_ER) {
+            at86rf2xx_spi_release_bus(dev);
+            return -EIO;
+        }
+        if (key) {
+            cfg = AT86RF2XX_AES_CTRL_AES_MODE__KEY |
+                  AT86RF2XX_AES_CTRL_AES_DIR__DEC;
+            _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+            _aes_transfer_bytes(dev, false, &cfg, NULL, sizeof(cfg));
+            _aes_open_read(dev, AT86RF2XX_REG__AES_KEY_START);
+            _aes_transfer_bytes(dev, false, NULL, key, AT86RF2XX_AES_KEY_LENGTH);
+        }
+        cfg = AT86RF2XX_AES_CTRL_AES_MODE__CBC |
+              AT86RF2XX_AES_CTRL_AES_DIR__ENC;
+        mirror = cfg | AT86RF2XX_AES_CTRL_MIRROR_AES_REQUEST__START;
+        for (unsigned i = 1; i < nblocks; i++) {
+            _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+            _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+            _aes_transfer_bytes(dev, true, plain[i], NULL, 1);
+            _aes_transfer_bytes(dev, true, plain[i] + 1,
+                                cipher ? cipher[i - 1] : NULL,
+                                AT86RF2XX_AES_BLOCK_SIZE - 1);
+            _aes_transfer_bytes(dev, false, &mirror,
+                                cipher ? cipher[i - 1] + AT86RF2XX_AES_BLOCK_SIZE - 1 : NULL,
+                                1);
+
+            xtimer_usleep(AT86RF2XX_AES_DELAY_US);
+            if ((status = _aes_check(dev)) & AT86RF2XX_AES_STATUS_MASK__AES_ER) {
+                at86rf2xx_spi_release_bus(dev);
+                return -EIO;
+            }
+        }
+
+        /* send dummy bytes to get the last block of cipher text */
+        uint8_t *mac = cipher ? cipher[nblocks - 1] : iv;
+        mirror &= ~AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_REQUEST;
+        _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+        _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+        _aes_transfer_bytes(dev, true, plain[0], NULL, 1);
+        _aes_transfer_bytes(dev, true, plain[0] + 1,
+                            mac, AT86RF2XX_AES_BLOCK_SIZE - 1);
+        _aes_transfer_bytes(dev, false, &mirror,
+                            mac + AT86RF2XX_AES_BLOCK_SIZE - 1, 1);
+
+        at86rf2xx_spi_release_bus(dev);
+    }
+    return status;
+}
+
+int at86rf2xx_aes_cbc_decrypt(at86rf2xx_t *dev,
+                              aes_block_ptr_t plain,
+                              uint8_t key[AT86RF2XX_AES_BLOCK_SIZE],
+                              uint8_t iv[AT86RF2XX_AES_BLOCK_SIZE],
+                              const aes_block_ptr_t cipher,
+                              uint8_t nblocks)
+{
+    uint8_t status = 0;
+    if (nblocks) {
```

Comment:
```suggestion
    if (!nblocks) {
        return 0;
    }
```

Reply:
I wonder if this is a matter of pipelining improvement or code style improvement?

Reply:
Code style - less deep indentations are generally easier to read.

If there is one large if block us humans have to keep that open curly bracket in mind, wondering what comes after the if, to reason about the code.
When the block is small, it's easier to immediately see what's going on.

The compiler won't care and generate the same code either way.